cmd/incusd/api_project.go

```diff
@@ -1541,3 +1566,93 @@ func projectValidateRestrictedSubnets(s *state.State, value string) error {
 
 	return nil
 }
+
+// Do I need to check the StatusCode and/or ExpandedConfig similar to "cmd/incus/delete.go"
+// Do I need to check if this a snapshot or of type container similar to "cmd/incusd/instance_delete.go"
+// Delete all project instances
+func projectDeleteInstances(ctx context.Context, project *cluster.Project, tx *db.ClusterTx) (error) {
+	instances, err := cluster.GetInstances(ctx, tx.Tx(), cluster.InstanceFilter{Project: &project.Name})
+	if err != nil {
+		return err
+	}
+
+	if len(instances) == 0 {
+		return nil
+	}
+
+	for _, instance := range instances {
+		if err := cluster.DeleteInstance(ctx, tx.Tx(), project.Name, instance.Name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// is there any value in checking the UsedBy ?
+// Delete all project profiles
+func projectDeleteProfiles(ctx context.Context, project *cluster.Project, tx *db.ClusterTx) (error) {
+	profiles, err := cluster.GetProfiles(ctx, tx.Tx(), cluster.ProfileFilter{Project: &project.Name})
+	if err != nil {
+		return err
+	}
+
+	if len(profiles) > 0 {
+		// Consider the project empty if it is only used by the default profile.
+		if len(profiles) == 1 && profiles[0].Name == "default" {
+			return nil
+		}
+		for _, profile := range profiles {
+			if profile.Name == "default" {
+				continue
+			}
+			if err := cluster.DeleteProfile(ctx, tx.Tx(), project.Name, profile.Name) ; err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
You don't need that much indentation.
You can probably just use:
```
if len(profiles) == 1 {
    return nil
}

for _, profile := range profiles {
    if profile.Name == "default" {
        continue
    }

    err := cluster.DeleteProfile(ctx, tx.Tx(), project.Name, profile.Name)
    if err != nil {
        return err
    }
}
```

Reply:
got it. made the change.